pkg/controllers/hypershiftdeployment_controller.go

```diff
@@ -176,6 +182,36 @@ func (r *HypershiftDeploymentReconciler) Reconcile(ctx context.Context, req ctrl
 	return ctrl.Result{}, nil
 }
 
+func (r *HypershiftDeploymentReconciler) setDefaultValueForHostedCluster(ctx context.Context, hyd *hypdeployment.HypershiftDeployment) error {
```

Comment:
nit, you may want to generate manifestwork for hostedCluster only when clusterID is set in hypershiftDeployment. That would help you to avoid additional update on manifestwork.

Also we may add more comment here on why clusterID must be set here. And indicate this should be reconsidered if manifestwork supports updateStrategy in the future.

Reply:
I feel like we want to put the ClusterID in the scaffold hostdClusterSpec so its just set, and if SD is creating this, they should just be required to include the ClusterID.
Same with OLMCatalogPlacement , what do you think?

I guess we could keep both, so that if SD leaves them out, then they get set.
Can you add both of these to the Scaffold hostedClusterSpec too?